proof-producer/bin/proof-producer/include/nil/proof-generator/commands/aggregated_fri_proof_command.hpp

```diff
@@ -0,0 +1,203 @@
+
+                    // create the transcript
+                    using transcript_hash_type = typename PlaceholderParams::transcript_hash_type;
+                    using transcript_type = crypto3::zk::transcript::fiat_shamir_heuristic_sequential<transcript_hash_type>;
+                    transcript_type transcript;
+
+                    transcript(aggregated_challenge.value());
+
+                    // Sum up all the polynomials from the files.
+                    polynomial_type sum_poly;
+                    for (const auto& path : input_combined_Q_polynomial_files) {
+                        std::optional<polynomial_type> next_combined_Q = PolynomialIO::read_poly_from_file(path);
+                        if (!next_combined_Q) {
+                            return CommandResult::UnknownError("Failed to read next combined Q from {}", path.string());
+                        }
+                        sum_poly += next_combined_Q.value();
+                    }
+                    auto lpc_scheme = commitment_scheme_fac_.make_lpc_scheme(table_description_->rows_amount);
+                    auto [fri_proof, challenges] = lpc_scheme->proof_eval_FRI_proof(sum_poly, transcript);
+
+                    // And finally run proof of work.
+                    ProofOfWork proof_of_work = nil::crypto3::zk::algorithms::run_grinding<FriType>(lpc_scheme->get_fri_params(), transcript);
+
+                    auto res = save_fri_proof_to_file(fri_proof, aggregated_fri_proof_output_file);
+                    if (!res) {
+                        return CommandResult::UnknownError("Failed to write aggregated FRI proof to file.");
+                    }
+
+                    res = save_proof_of_work(proof_of_work, proof_of_work_output_file);
+                    if (!res) {
+                        return CommandResult::UnknownError("Failed to write proof of work to file.");
+                    }
+
+                    res = ChallengeIO::save_challenge_vector_to_file(challenges, consistency_checks_challenges_output_file);
+                    if (!res) {
+                        return CommandResult::UnknownError("Failed to write consistency checks challenges to file.");
+                    }
+
+                    return CommandResult::Ok();
+                }
+
+            private:
+                std::shared_ptr<TableDescription> table_description_;
+
+                CommitmentSchemeFac commitment_scheme_fac_;
+
+                boost::filesystem::path aggregated_challenge_file_;
+                std::vector<boost::filesystem::path> input_combined_Q_polynomial_files_;
+                boost::filesystem::path aggregated_fri_proof_output_file_;
+                boost::filesystem::path proof_of_work_output_file_;
+                boost::filesystem::path consistency_checks_challenges_output_file_;
+            };
+
+        template <typename CurveType, typename HashType>
+        struct AggregatedFriProofCommand: public command_chain {            
```

Comment:
Some commands derive from chain, some others from command_step itself. Maybe we derive them all from chain?

I think if we want to be extremely picky, we could create not a chain but a tree of commands, so each command will know its subcommands which it needs to perform to finish the operation, something like the "Composite Pattern".